.github/workflows/pre-pr-checks.yml

```
@@ -0,0 +1,87 @@
+          ruff format . --quiet || true
+          ruff check .
+
+      - name: Type check (pyright)
+        run: pyright
+
+      - name: Run tests
+        run: |
+          pytest agents/tests/ -v --cov=agents.governance --cov=agents.core_logic --cov-report=term-missing --cov-fail-under=45
+
+      - name: Codacy analysis (required when configured)
+        if: ${{ secrets.CODACY_TOKEN != '' }}
```

actionlint (1.7.9)

context "secrets" is not allowed here. available contexts are "env", "github", "inputs", "job", "matrix", "needs", "runner", "steps", "strategy", "vars". see https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability for more details

(expression)